content/channels/index.textile

```diff
@@ -999,6 +999,148 @@ channelMessageSubscription.cancel();
+```
+
+The following is an example of a filter expression subscribing to messages with a flavor of either “strawberry” or “chocolate”:
+
+```[text]
+headers.flavor == `"strawberry"` || headers.flavor == `"chocolate"`
+```
+
+h4(#filter-subscribe). Subscribe with a filter
+
+In order to use a subscribe to a channel with a filter expression, you obtain a channel instance using the @getDerived()@ method. This accepts a filter expression as a parameter. 
+
+The following is an example of subscribing to a channel using one of the previous example filters:
+
+```[realtime_javascript]
+realtime.channels.getDerived('scoops-kiosk', {
+    filter: 'name == `"ice-cream"` && headers.flavor == `"strawberry"` && headers.cost < `50`'
+}).subscribe(...);
+```
+
+<aside data-type="note">
+<p>Clients that are publishing to the same channel that they are subscribed to using a filter need to obtain a channel instance twice. Once with the filter expression using @getDerived()@ for the subscription and once using "@get()@":#create for publishing. Attempts to publish to a channel created or retrieved with a filter expression will fail.</p>
+</aside>
+
+The following example demonstrates publishing to a channel, but subscribing to only a subset of messages on it:
+
+```[realtime_javascript]
+// Connect to Ably
+const realtime = new Ably.Realtime({'{{API_KEY}}'});
+
+// Create a channel instance to publish to
+const pubChannel = realtime.channels.get('scoops-kiosk');
+
+// Create a channel instance using the filter qualifier
+const subChannel = realtime.channels.getDerived('scoops-kiosk', {
+   filter: 'name == `"ice-cream"` && headers.flavor == `"strawberry"` && headers.cost < `50`'
+});
+
+// Subscribe to the channel using the filtered subscription
+subChannel.subscribe((message) => {
+   alert('Ice cream update: ' + message.data);
+ });
+
+// Publish to the unfiltered channel instance
+pubChannel.publish({
+   name: 'ice-cream',
+   data: '...',
+   extras: {
+       headers: {
+           flavor: "strawberry",
+           cost: 35,
+           temp: 3
+       }
+   });
+});
+```
+
+h4(#filter-capabilities). Subscription filter capabilities
+
+Clients require the subscribe "capability":/auth/capabilities for one of the following resources in order to receive messages from a subscription filter:
+
+* @[filter]<channel name>@
+* @[*]<channel name>@
+* @[*]*@
+
+It is important to set the correct capabilities for clients that will be carrying out additional operations on the same channel they are subscribed to using a filter. If clients have the subscribe capability for the channel itself and attach to it, such as when entering the presence set, they will be streamed all messages.
+
+The following is an example of a request for a token with only the subscribe capability for the subscription filter, and only the publish capability for the unfiltered channel. This avoids the client from being streamed all messages, even if they attach to the unfiltered channel:
```

Comment:
I don't think this is strictly true. If you specify a filter that matches all messages (i.e. suppresses nothing), then you still receive all messages. The only impact of the capability is that it is required in order to make a filtered subscription, but itself does not determine the set of messages received.

Reply:
Maybe I've misunderstood here (and I should write like the following where you specify the actual filter in the capability):

`[<filter>]<channel-name>`

In which case, if you then have `subscribe` capabilities to the channel separately (unfiltered), for example to receive presence updates, then you'd be streamed all messages at the point of attaching to the channel, no?

Reply:
So currently we don't have support in capabilities for a specific filter expression. This is something we have done some design & thinking around (see [here](https://ably.atlassian.net/wiki/spaces/product/pages/2607546413/DSDR3+Long+Form+Capabilities) and [here](https://ably.atlassian.net/wiki/spaces/product/pages/2607087679/DSDR4+Filtered+Channel+Capabilities)), and we intend to support it as part of the GA implementation, but it's not currently supported in the preview. For now, the only thing that's necessary is the `[filter]*` capability, where `filter` refers to the `qualifier`, not the `qualifierParam` (i.e. the filter expression). More info here: https://ably.atlassian.net/wiki/spaces/ENG/pages/2508619954/ADR82+Filtered+Subscriptions#Capabilities

Reply:
I think I may need to reword the paragraph and add a better example, as I was trying to approach this from a 'messages saved' point of view. 

Say for example I have the following capabilities: 

```plaintext
   "[filter]scoops-kiosk": ["subscribe"],
   "scoops-kiosk": ["subscribe"]
```

If I subscribe with a filter to the channel `scoops-kiosk`, that attaches me to the derived channel. As it's a derived channel I won't be sent all messages, as we apply server-side filtering (unless my filter happens to match all messages).

If I then subscribe to `scoops-kiosk` (underived), for example to register a listener for presence events. Would I be streamed all regular message events from the channel? If so, I'm filtering the content I receive, but I'm not saving on message counts.

Reply:
Rewritten this section to be simpler, and hopefully clearer.